habitat-lab/habitat/datasets/rearrange/rearrange_generator.py

```diff
@@ -459,56 +460,63 @@ def generate_episodes(
         """
         generated_episodes: List[RearrangeEpisode] = []
         failed_episodes = 0
+        total_episodes = 0
         if verbose:
             pbar = tqdm(total=num_episodes)
         while len(generated_episodes) < num_episodes:
+            logger.info(f"Generating... succeeded / total: {len(generated_episodes)} / {total_episodes}")
+            total_episodes += 1
             try:
                 self._scene_sampler.set_cur_episode(len(generated_episodes))
                 new_episode = self.generate_single_episode()
+                if new_episode is None:
+                    continue
                 rigid_objs = new_episode.rigid_objs
                 target_objs = new_episode.targets
+                if resume is not None:
+                    cameras_info = resume['perception']['cameras_info']
+                    arm_workspace = resume['manipulation']['arm_workspace']
+                    mobility_summary = resume['mobility']['summary']
 
-                cameras_info = resume['perception']['cameras_info']
-                arm_workspace = resume['manipulation']['arm_workspace']
-                mobility_summary = resume['mobility']['summary']
-
-                # TODO(ZXZ): modify episode generation algorithm
+                # TODO(zxz): modify episode generation algorithm
```

Comment:
Does this mean you want to save the episode robot resume information when generating the episode? Can you explain why?

Reply:
This is just for the optional argument `--resume` stated at https://github.com/SgtVincent/habitat-lab/blob/73eec9937cdde76669edfd475e794e4384574b77/habitat-mas/README.md?plain=1#L105. Since for now, I didn't use robot resume for generation, which is only based on some handmade magic number. This is for that we can still process the dataset generation even if there is no resume config.